src/lsp/main.ts

```diff
@@ -431,7 +430,14 @@ async function startClientCommand() {
     await startClient();
 }
 
-async function startClient(): Promise<void> {
+async function startClient(): Promise<boolean> {
+    const runningClient = getStateValue(LSP_CLIENT_KEY);
+    if (runningClient) {
+        await runningClient.stop().catch((e) => {
+            console.log('Error stopping running client:', e);
+        });
+    }
```

Comment:
Is this a thing we're running into, where there is already a running client and we need to start a new one?

Reply:
Astute observation! What happens right now is that if we stop the client when we notice it is not working, the user gets an error alert. (See separate TODO comment). So we let it run. The user I have in mind here is the beginner, using the getting started repl. Showing that alert could make things seem much more broken then they are.

Reply:
I added a comment here as well, informing why we have this `runningClient` check.